api/server.go

```diff
@@ -78,7 +79,10 @@ func (c apiServer) Start(ctx context.Context, wg *sync.WaitGroup, startupErr cha
 		mux.Handle("/debug/", http.DefaultServeMux)
 	}
 
-	mux.Handle("/", gwmux)
+	// if we dont find the route in the mux, we will redirect it to gmux
```

Comment:
How would an interceptor set up for authentication look like?

Reply:
For now the, since ping doesn't have any auth there is no interceptor and for the remaining, it will happen as part of the `gwmux`. Once we move to this framework, we will have a middleware just like the `middleware.Recovery` i have defined which exclude `ping`

Reply:
Yep, I understand that in this particular endpoint we exclude the `auth`. My question is more on looking ahead of this endpoint into a `create` for example. That requires authentication, How would it look like in the code? Can you add a simple snippet here as a comment?

Reply:
Not sure I understand what you are asking here :-D